apps/remote_control/lib/lexical/remote_control/build/state.ex

```diff
@@ -50,10 +55,15 @@ defmodule Lexical.RemoteControl.Build.State do
   end
 
   def compile_project(%__MODULE__{} = state, force?) do
+    state = increment_build_number(state)
     project = state.project
 
     Build.with_lock(fn ->
-      {elapsed_us, result} = :timer.tc(fn -> safe_compile_project(state.project, force?) end)
+      compile_requested_message =
+        project_compile_requested(project: project, build_number: state.build_number)
```

Comment:
They won't though, again, there's no overwriting, only appending, unless the build number increments, and then they replace what's there. 
The reason that we don't need the build number in the project compiled message is that we've already received it in the `project_compile_requested` message. Despite this looking like a distributed system, it's really not one, so the messages are always in order, so you'll always get the `project_compile_requested` before the `project_compiled` message, and you can plan accordingly.

Reply:
One thing I had thought of doing is making plugin specific messages, that will make clearing things slightly simpler, but I didn't find any bugs with the current method.

Reply:
You can replicate this bug by creating a function like the one below:

```elixir
defmodule WithCredoPlugin do
  @moduledoc false
  def foo() do
    f
  end
end
```

And the diagnostics should look like 

<img width="364" alt="image" src="https://github.com/lexical-lsp/lexical/assets/12830256/4fd9543e-91ef-43d7-8991-9d3cc65fc01d">

But when you save that file, it will become 

<img width="324" alt="image" src="https://github.com/lexical-lsp/lexical/assets/12830256/dcc93803-d5c4-41d8-a592-9f2aa63be512">

the diagnostics from `Mix` disappeared

Reply:
Alright, figured this out. It should be fixed now. Unresolve this if you think it's broken.